src/uct/ib/mlx5/rc/rc_mlx5.inl

```diff
@@ -887,30 +889,33 @@ void uct_rc_mlx5_txqp_dptr_post_iov(uct_rc_mlx5_iface_common_t *iface, int qp_ty
 #if HAVE_MLX5_MMO
     case MLX5_OPCODE_MMO|UCT_RC_MLX5_OPCODE_FLAG_MMO_PUT:
     case MLX5_OPCODE_MMO|UCT_RC_MLX5_OPCODE_FLAG_MMO_GET:
-        gga_wqe                  = txwq->curr;
+        dma_seg                  = UCS_PTR_TYPE_OFFSET(ctrl, *ctrl);
 #if ENABLE_DEBUG_DATA
-        gga_wqe->padding         = 0xdeadbeef;
+        dma_seg->padding         = 0xdeadbeef;
 #endif
-        gga_wqe->be_opaque_lkey  = htobe32(opaque_mr->lkey);
-        gga_wqe->be_opaque_vaddr = htobe64((uintptr_t)opaque_mr->addr);
+        dma_seg->be_opaque_lkey  = opaque_mr->be_lkey;
+        dma_seg->be_opaque_vaddr = opaque_mr->be_vaddr;
 
-        ucs_assert(iovcnt == 1);
+        ucs_assertv(iovcnt == 1, "iovcnt %zu is not supported by MMO", iovcnt);
         if (opcode_flags & UCT_RC_MLX5_OPCODE_FLAG_MMO_PUT) {
-            uct_ib_mlx5_set_data_seg(&gga_wqe->gather, iov[0].buffer,
-                                     iov[0].length,
-                                     uct_ib_memh_get_lkey(iov[0].memh));
-            uct_ib_mlx5_set_data_seg(&gga_wqe->scatter,
-                                     (const void*)remote_addr, iov[0].length,
-                                     rkey);
+            dma_src_buf = iov[0].buffer;
+            dma_src_key = uct_ib_memh_get_lkey(iov[0].memh);
+            dma_dst_buf = (const void*)remote_addr;
+            dma_dst_key = rkey;
         } else /* UCT_RC_MLX5_OPCODE_FLAG_MMO_GET */ {
-            uct_ib_mlx5_set_data_seg(&gga_wqe->gather, (const void*)remote_addr,
-                                     iov[0].length, rkey);
-            uct_ib_mlx5_set_data_seg(&gga_wqe->scatter, iov[0].buffer,
-                                     iov[0].length,
-                                     uct_ib_memh_get_lkey(iov[0].memh));
+            dma_src_buf = (const void*)remote_addr;
+            dma_src_key = rkey;
+            dma_dst_buf = iov[0].buffer;
+            dma_dst_key = uct_ib_memh_get_lkey(iov[0].memh);
         }
 
-        wqe_size = sizeof(*gga_wqe);
+        dptr = UCS_PTR_TYPE_OFFSET(dma_seg, *dma_seg);
```

Comment:
shall we use uct_ib_mlx5_txwq_wrap_exact ?

Reply:
why? can single WQE be wrapped by segments?

Reply:
so pls use uct_ib_mlx5_txwq_wrap_none